### PR TITLE
start progress on #9 by adding yarn and npm TypeDoc dependencies

### PR DESCRIPTION
**Pull request by hirehamir**
Started working on #9. I did run into a weird error message, though.

I try to output documentation to the default directory using `npx typedoc --out`, and it seems to work, but the terminal gives me the following error message:

```
Using TypeScript 4.0.2 from /Users/hamirmahal/OneDrive - sjsu.edu/repositories/react-wavesurfer.js/node_modules/typescript/lib
No 'out' or 'json' option has been set
The './docs' directory has been set as the output location by default
Rendering [=======---------------------------------] 16%Could not find the language 'tsx', did you forget to load/include a language module?
Warning: Unknown language: "tsx"
Could not find the language 'tsx', did you forget to load/include a language module?
Warning: Unknown language: "tsx"
Could not find the language 'tsx', did you forget to load/include a language module?
Warning: Unknown language: "tsx"
Rendering [========================================] 100%

Documentation generated at /Users/hamirmahal/OneDrive - sjsu.edu/repositories/react-wavesurfer.js/docs
```

After this, if we figure out how to host TypeDoc along with storybook, I think we'll be done with #9.

[I read that TypeDoc already added support for `.tsx` files](https://github.com/TypeStrong/typedoc/issues/147), so I was surprised to see that error message.

### CODE REVIEW

File: package.json
Comment (amilajack) [inline package.json]: We should remove the generated files from the PR with `rm -rf docs`. They should still be in version control beceause they'll be added in `gh-pages` branch. Also here's an example of what the npm script should look like:

https://github.com/amilajack/joker/blob/master/package.json#L19

Afte adding this, we can build the docs like so: `yarn docs`

**Review by amilajack — CHANGES_REQUESTED** [review]

File: package.json
Comment (amilajack) [reply package.json]: Also while we're at it we might as well add a `deploy` npm script similar to https://github.com/amilajack/freesound-player/blob/master/package.json#L47. But rather than building we'd want to build the docs.

**Review by amilajack — CHANGES_REQUESTED** [review]

File: package.json
Comment (hirehamir) [reply package.json]: [`"docs": "rm -rf docs && typedoc ./src --out docs",`](https://github.com/amilajack/joker/blob/110246c98e90de6565400547847bb2235951faf6/package.json#L19)

Small change, but we can remove ` docs` after `typedoc ./src --out` from `"docs": "rm -rf docs && typedoc ./src --out docs",` because the default output directory of TypeDoc is `docs` anyway, right?

Comment (hirehamir) [reply package.json]: > Also while we're at it we might as well add a `deploy` npm script similar to https://github.com/amilajack/freesound-player/blob/master/package.json#L47. But rather than building we'd want to build the docs.

`"deploy": "yarn docs && gh-pages -d build",` is what I have for this; is it correct?

**Review by hirehamir — COMMENTED** [review]
So, I think I'm just adding

```
    "deploy": "yarn docs && gh-pages -d build",
    "docs": "rm -rf docs && typedoc ./src --out",
```

**Comment by hirehamir** [comment]
@amilajack To resolve conflicts, I was thinking

```
    "build": "rm -rf dist && tsc && yarn build-storybook",
    "deploy": "yarn docs && gh-pages -d build",
    "docs": "rm -rf docs && typedoc ./src --out",
```

instead of `"build": "rm -rf dist && tsc",`

Does that work? I think I can only `Commit merge` after I `Resolve conflicts`, so I wanted to get approval before merging, unlike I did last time.

File: package.json
Comment (amilajack) [reply package.json]: Actually my bad for backtracking but now that I think about it, if we deploy both storybook and typedoc, the one published second to gh pages will override what was published previously

Comment (amilajack) [reply package.json]: Let's take out deploy script for now.

File: package.json
Comment (amilajack) [inline package.json]: Since `storybook-to-ghpages` is going to publish to gh pages, this command will "double publish" to gh pages. For now I think we should remove this.

**Review by amilajack — CHANGES_REQUESTED** [review]

File: package.json
Comment (hamirmahal) [reply package.json:51]: So, instead of

`"deploy": "yarn docs && gh-pages -d build",`

it should just say

`"deploy": "yarn docs",`?

Comment (amilajack) [reply package.json]: yea sounds good

Comment (amilajack) [reply package.json]: yea sounds good

Comment (hirehamir) [reply package.json]: Okay, I changed it from 

`"deploy": "yarn docs && gh-pages -d build",`

to

`"deploy": "yarn docs",`

**Review by hirehamir — COMMENTED** [review]
Creating a commit to change

`"deploy": "yarn docs && gh-pages -d build",`

to

`"deploy": "yarn docs",`

**Review by amilajack — CHANGES_REQUESTED** [review]
I think you might have accidentally added changes from previous commits. Try resetting your branch to be equivalent to master and adding the changes again:

```bash
git checkout add-typedoc-support # make sure you're on your branch
git pull --all
git reset --hard origin/master
```

File: package.json
Comment (amilajack) [inline package.json]:
```suggestion

```

**Review by amilajack — CHANGES_REQUESTED** [review]

**Comment by amilajack** [comment]
Yessss!

**Pull request merged**
